hypha/apply/activity/services.py

```diff
@@ -37,44 +39,39 @@ def edit_comment(activity: Activity, message: str) -> Activity:
     return activity
 
 
-def get_related_actions_for_user(obj, user):
-    """Return Activity objects related to an object, esp. useful with
-    ApplicationSubmission and Project.
-
-    Args:
-        obj: instance of a model class
-        user: user who these actions are visible to.
-
-    Returns:
-        [`Activity`][hypha.apply.activity.models.Activity] queryset
-    """
-    related_query = type(obj).activities.rel.related_query_name
-
-    return (
-        Activity.actions.filter(**{related_query: obj})
-        .select_related("user")
-        .prefetch_related(
-            "related_object",
-        )
-        .visible_to(user)
-    )
-
-
-def get_related_comments_for_user(obj, user):
+def get_related_activities_for_user(obj, user):
     """Return comments/communications related to an object, esp. useful with
     ApplicationSubmission and Project.
 
     Args:
-        obj: instance of a model class
+        obj: instance of either an [`ApplicationSubmission`][hypha.apply.funds.models.submissions.ApplicationSubmission] or [`Project`][hypha.apply.projects.models.project.Project].
         user: user who these actions are visible to.
 
     Returns:
         [`Activity`][hypha.apply.activity.models.Activity] queryset
     """
-    related_query = type(obj).activities.rel.related_query_name
+    proj_content_type = ContentType.objects.get_for_model(Project)
+    app_content_type = ContentType.objects.get_for_model(ApplicationSubmission)
+
+    # Determine if the provided object is an ApplicationSubmission or Project, then pull
+    # the related it's related Project/ApplicationSubmission activites if attribute it exists
+    if isinstance(obj, ApplicationSubmission):
+        source_filter = Q(source_object_id=obj.id, source_content_type=app_content_type)
+        if hasattr(obj, "project") and obj.project:
+            source_filter = source_filter | Q(
+                source_object_id=obj.project.id, source_content_type=proj_content_type
+            )
+    else:
+        source_filter = Q(
+            source_object_id=obj.id, source_content_type=proj_content_type
+        )
+        if hasattr(obj, "submission") and obj.submission:
+            source_filter = source_filter | Q(
+                source_object_id=obj.submission.id, source_content_type=app_content_type
+            )
```

Comment:
I believe this can further simplified like this, 

logic to get the source filter. 
```python
    if hasattr(obj, "project") and obj.project:
        source_filter = Q(submission=obj) | Q(project=obj.project)
    if hasattr(obj, "submission") and obj.submission:
        source_filter = Q(submission=obj.submission) | Q(project=obj)
    else:
        source_filter = Q(submission=obj)
```

```suggestion
    if hasattr(obj, "project") and obj.project:
        source_filter = Q(submission=obj) | Q(project=obj.project)
    if hasattr(obj, "submission") and obj.submission:
        source_filter = Q(submission=obj.submission) | Q(project=obj)
    else:
        source_filter = Q(submission=obj)

```

Reply:
this worked! when I tried this before but I kept running into issues with the GenericForeignKey, not sure what I was trying before

Reply:
👍🏼